src/Renci.SshNet/Security/Cryptography/ED25519Key.cs

```diff
@@ -91,8 +92,7 @@ public ED25519Key(SshKeyData publicKeyData)
                 throw new ArgumentException($"Invalid Ed25519 public key data ({publicKeyData.Name}, {publicKeyData.Keys.Length}).", nameof(publicKeyData));
             }
 
-            PublicKey = publicKeyData.Keys[0].ToByteArray().Reverse().TrimLeadingZeros().Pad(Ed25519.PublicKeySizeInBytes);
-            PrivateKey = new byte[Ed25519.ExpandedPrivateKeySizeInBytes];
+            PublicKey = publicKeyData.Keys[0].ToByteArray().Reverse().TrimLeadingZeros().Pad(Ed25519.PublicKeySize);
```

Comment:
I'd just keep the PrivateKey not-null unless we enable nullability here

```suggestion
            PublicKey = publicKeyData.Keys[0].ToByteArray().Reverse().TrimLeadingZeros().Pad(Ed25519.PublicKeySize);
            PrivateKey = new byte[Ed25519.SecretKeySize];
```